### PR TITLE
Fix #8614

### LINKED ISSUE #8614: Implementing EXT_texture_webp to gltf loader
Since we are using 3D in web, the smaller the file size the better. Unfortunately 2k, 4k, 8k textures are very big using png or jpg.

I would love to use [EXT_texture_webp](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/EXT_texture_webp) with babylon js because webp is able to compress the file size of textures massively.

Forum: https://forum.babylonjs.com/t/using-webp-with-gltf/12732

### PR DESCRIPTION
**Pull request by deltakosh**
Adding support for EXT_texture_webp

### CODE REVIEW

**Comment by Popov72** [comment]
Shouldn't you check if the browser supports .webp pictures and fallback using the regular image if not (and if provided in the gltf)?

Or do we assume webp support is wide enough and it is not necessary?

File: loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { BaseTexture } from "babylonjs/Materials/Textures/baseTexture";
+import { Nullable } from "babylonjs/types";
+import { IKHRTextureBasisU } from 'babylonjs-gltf2interface';
+
+const NAME = "EXT_texture_webp";
+
+/**
+ * [Proposed Specification](https://github.com/KhronosGroup/glTF/pull/1534)
+ * !!! Experimental Extension Subject to Changes !!!
+ */
```
Comment (Popov72) [inline loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts:13]: As I can see it, the extension is now live on Khronos side (https://github.com/KhronosGroup/glTF/blob/master/extensions/README.md)

**Comment by deltakosh** [comment]
The latter:
image

Only IE11 is a problem and I'm slowly moving to mental model where new feature should not be slowed down by IE anymore ;)

File: loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts (hunk 1 above)
Comment (deltakosh) [reply loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts:13]: oh yeah..even better ;D

File: loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        delete this._loader;
+    }
+
+    /** @hidden */
+    public _loadTextureAsync(context: string, texture: ITexture, assign: (babylonTexture: BaseTexture) => void): Nullable<Promise<BaseTexture>> {
+        return GLTFLoader.LoadExtensionAsync<IEXTTextureWebP, BaseTexture>(context, texture, this.name, (extensionContext, extension) => {
+            const sampler = (texture.sampler == undefined ? GLTFLoader.DefaultSampler : ArrayItem.Get(`${context}/sampler`, this._loader.gltf.samplers, texture.sampler));
+            const image = ArrayItem.Get(`${extensionContext}/source`, this._loader.gltf.images, extension.source);
+            return this._loader._createTextureAsync(context, sampler, image, (babylonTexture) => {
+                babylonTexture.gammaSpace = false;
```
Comment (sebavan) [inline loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts:39]: Just wondering about this one ???

Comment (deltakosh) [reply loaders/src/glTF/2.0/Extensions/EXT_texture_webp.ts:39]: Totally right! I'll remove it and merge
Thanks buddy!

**Comment by bghgary** [comment]
Might be good to add a test for this some day :)

**Comment by deltakosh** [comment]
It is there already

**Comment by bghgary** [comment]
Yes, yes, it is. 😄

**Pull request merged**